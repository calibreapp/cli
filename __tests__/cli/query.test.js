const {
  runCLI,
  setupIntegrationServer,
  teardownIntegrationServer
} = require('../utils')
const site = require('../fixtures/site.json')

describe('query', () => {
  beforeAll(() => setupIntegrationServer(site))
  afterAll(() => teardownIntegrationServer())

  test('missing query argument', () => {
    return runCLI({ args: 'query', testForError: true }).then(stdout => {
      expect(stdout).toMatchSnapshot()
    })
  })

  it('returns response', () => {
    const slug = 'calibre'
    return runCLI({
      args: `query --query='query GetSite($slug: String!) {organisation{site(slug: $slug){slug}}}' --slug=${slug}`
    }).then(stdout => {
      expect(stdout).toMatchSnapshot()
    })
  })
})
