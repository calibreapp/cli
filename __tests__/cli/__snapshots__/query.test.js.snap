// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query missing query argument 1`] = `
"

 Missing required argument: query 

 --- Command documentation
 cli.js query

Query the Calibre GraphQL API

Options:

  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --query      GraphQL query to execute                      [string] [required]
  --variables  Pass query variables as named arguments

Examples:
  calibre query --query='query GetSite($slug: String!)
  {organisation{site(slug: $slug){slug}}}' --slug=calibre
"
`;

exports[`query returns response 1`] = `
"{ organisation: { site: { slug: 'calibre' } } }
"
`;
